client/js/widgets/widget.js

```diff
@@ -1180,6 +1171,36 @@ export class Widget extends StateManaged {
     await this.updatePiles();
   }
 
+  async moveFlipToHolder(children, source, target, count, face) {
+    for (const c of children.slice(0, count || 999999).reverse()) {
+      if(face !== null && c.flip)
+        await c.flip(face);
+      let isInTargetHolder = (source == target);
+      if (!source && c.parent !== undefined) {
+        if (c.parent.get('type') == 'pile' && c.parent.parent !== undefined)
+          isInTargetHolder = (c.parent.parent == target);
```

Comment:
You shouldn't use `c.parent` here. I'm pretty sure it's not what you want. It's not always the same as `widgets.get(c.get('parent'))` which is probably what you need here.